R/cohenFleissKappa.R

```diff
@@ -0,0 +1,202 @@
+#
+# Copyright (C) 2021 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+cohenFleissKappa <- function(jaspResults, dataset, options) {
+  
+  ready <- length(options[["variables"]]) > 1
+  
+  dataset <- .cohensFleissKappaReadData(dataset, options)
+    
+  if(options[["cohensKappa"]])
+    jaspResults[["cohensKappa"]] <- .cohensKappa(dataset, options, ready)
+  
+  if(options[["fleissKappa"]])
+    jaspResults[["fleissKappa"]] <- .fleissKappa(dataset, options, ready)
+  
+  return()
+}
+
+# Read in the dataset (copied from .reliabilityReadData)
+.cohensFleissKappaReadData <- function(dataset, options) {
+  variables <- unlist(options[["variables"]])
+  if (is.null(dataset)) {
+    dataset <- .readDataSetToEnd(columns.as.factor = variables)
+  }
+  return(dataset)
+}
+
+.cohensKappa <- function(dataset, options, ready){
+  
+  weighted <- options[["cohensWeightedOrNot"]] == "cohensWeighted"
+  
+  if(weighted){
+    weightedString <- "Weighted"
+  }else{
+    weightedString <- "Unweighted"
+  }
+  
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Cohen's %s Kappa", casefold(weightedString, upper = F)))
+  jaspTable$addColumnInfo(name = "ratings", title = gettext("Ratings"), type = "string")
+  jaspTable$addColumnInfo(name = "cKappa", title = gettextf("%s Kappa", weightedString), type = "number")
+  jaspTable$position <- 1
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "cohensKappa",
+      "cohensWeightedOrNot",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+  
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  
+  if(ready){
```

Comment:
```suggestion
  if (ready) {
```
There should be a space before the if condition, also throughout